File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<>();
        
        for (String name : participant){
            map.put(name, map.getOrDefault(name, 0) +1);
        }
        for (String name : completion){
            map.put(name, map.get(name) -1);
        }
//         for (Map.Entry<String, Integer> entry : map.entrySet()){
//             if (entry.getValue() > 0)
//                 answer = entry.getKey();
//         }
        for (String key : map.keySet()){
            if (map.get(key) > 0){
                answer = key;
            }
        }
        
        return answer;
    }
}
// 해시값으로 participant 돌면서 추가, completion으로 돌면서 -1,
// value값이 0 이상인거가 답.(마이너스거나 양수인 키가 2개 이상이면 오류)

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    # participant으로 값 0으로 통일한 해시만들기(키값 같으면 +1+1)
    # completion돌며 존재하는 값 찾아서 -1
    # for문으로 해시 돌며 찾으면 0 아닌값 찾기
    dict = {}
    
    for i in participant:
        if i not in dict:
            dict[i] = 0
        dict[i] += 1
        
    for i in completion:
        dict[i] -= 1    
        
    for key, values in dict.items():
        if values != 0:
            return key
    return answer